src/validation.cpp

```diff
@@ -499,15 +500,22 @@ static bool AcceptToMemoryPoolWorker(const CChainParams& chainparams, CTxMemPool
                 // first-seen mempool behavior should be checking all
                 // unconfirmed ancestors anyway; doing otherwise is hopelessly
                 // insecure.
+                //
+                // All transactions in mempool become replaceable after the timeout.
                 bool fReplacementOptOut = true;
                 if (fEnableReplacement)
                 {
-                    for (const CTxIn &_txin : ptxConflicting->vin)
+                    const int64_t nConflictingTime = pool.info(ptxConflicting->GetHash()).nTime;
+                    fReplacementOptOut = nAcceptTime - nConflictingTime < nReplacementTimeout;
+                    if (fReplacementOptOut)
                     {
-                        if (_txin.nSequence < std::numeric_limits<unsigned int>::max()-1)
+                        for (const CTxIn &_txin : ptxConflicting->vin)
```

Comment:
I only moved the code but indeed that's a good idea (addressed in #38f4c85b47351d40700e088122317aa385c5ee65 )